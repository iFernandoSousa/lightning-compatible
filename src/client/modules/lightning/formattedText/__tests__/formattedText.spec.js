/*
 * Copyright (c) 2019, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: MIT
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
 */

import { createElement } from 'lwc';
import Element from 'c/formattedText';

const createComponent = () => {
    const element = createElement('c-formatted-text', { is: Element });
    document.body.appendChild(element);
    return element;
};
const exampleText =
    'I love www.salesforce.com \n And I like cats@salesforce.com';

describe('c-formatted-text', () => {
    it('default', () => {
        const element = createComponent();
        expect(element).toMatchSnapshot();
    });

    it('parses multiple new lines into multiple BRs', () => {
        const element = createComponent();
        element.value = 'First line \r\n\r\n Fourth line \n\n\n Eighth line';
        element.linkify = true;
        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    it('parses url after ?', () => {
        const element = createComponent();
        element.value = 'First http://www.some-business.com?a=1&b=2';
        element.linkify = true;
        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    it('with linkified text', () => {
        const element = createComponent();
        element.value = exampleText;
        element.linkify = true;
        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    it('without linkified text', () => {
        const element = createComponent();
        element.value = exampleText;
        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });
});

describe('c-formatted-text edge cases', () => {
    it('url only', () => {
        const element = createComponent();
        element.value = 'www.salesforce.com';
        element.linkify = true;
        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    it('newline only', () => {
        const element = createComponent();
        element.value = '\r\n';
        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    it('empty value', () => {
        const element = createComponent();
        element.value = null;
        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    it('empty text', () => {
        const element = createComponent();
        element.value = '';
        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    it('url and newline', () => {
        const element = createComponent();
        element.value = 'www.salesforce.com\r\n';
        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });
});
