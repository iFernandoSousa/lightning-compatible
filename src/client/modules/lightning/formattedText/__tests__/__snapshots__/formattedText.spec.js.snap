// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-formatted-text default 1`] = `
<c-formatted-text>
  #shadow-root(open)
</c-formatted-text>
`;

exports[`c-formatted-text edge cases empty text 1`] = `
<c-formatted-text>
  #shadow-root(open)
</c-formatted-text>
`;

exports[`c-formatted-text edge cases empty value 1`] = `
<c-formatted-text>
  #shadow-root(open)
</c-formatted-text>
`;

exports[`c-formatted-text edge cases newline only 1`] = `
<c-formatted-text>
  #shadow-root(open)
    
    <br />
    
</c-formatted-text>
`;

exports[`c-formatted-text edge cases url and newline 1`] = `
<c-formatted-text>
  #shadow-root(open)
    www.salesforce.com
    <br />
    
</c-formatted-text>
`;

exports[`c-formatted-text edge cases url only 1`] = `
<c-formatted-text>
  #shadow-root(open)
    <a
      href="http://www.salesforce.com"
      rel="noopener"
      target="_blank"
    >
      www.salesforce.com
    </a>
</c-formatted-text>
`;

exports[`c-formatted-text parses multiple new lines into multiple BRs 1`] = `
<c-formatted-text>
  #shadow-root(open)
    First line 
    <br />
    <br />
     Fourth line 
    <br />
    <br />
    <br />
     Eighth line
</c-formatted-text>
`;

exports[`c-formatted-text parses url after ? 1`] = `
<c-formatted-text>
  #shadow-root(open)
    First 
    <a
      href="http://www.some-business.com?a=1&b=2"
      rel="noopener"
      target="_blank"
    >
      http://www.some-business.com?a=1&b=2
    </a>
</c-formatted-text>
`;

exports[`c-formatted-text with linkified text 1`] = `
<c-formatted-text>
  #shadow-root(open)
    I love 
    <a
      href="http://www.salesforce.com"
      rel="noopener"
      target="_blank"
    >
      www.salesforce.com
    </a>
     
    <br />
     And I like 
    <a
      href="mailto:cats@salesforce.com"
      rel="noopener"
      target="_blank"
    >
      cats@salesforce.com
    </a>
</c-formatted-text>
`;

exports[`c-formatted-text without linkified text 1`] = `
<c-formatted-text>
  #shadow-root(open)
    I love www.salesforce.com 
    <br />
     And I like cats@salesforce.com
</c-formatted-text>
`;
